rust/kcl-lib/src/std/extrude.rs

```diff
@@ -412,23 +437,73 @@ pub(crate) async fn do_post_extrude<'a>(
     }
 
     let Faces {
-        sides: face_id_map,
+        sides: mut face_id_map,
         start_cap_id,
         end_cap_id,
     } = analyze_faces(exec_state, args, face_infos).await;
+
+    // If this is a clone, we will use the clone_id_map to map the face info from the original sketch to the clone sketch.
+    if sketch.clone.is_some() && clone_id_map.is_some() {
+        face_id_map = face_id_map
+            .into_iter()
+            .filter_map(|(k, v)| {
+                let fe_key = clone_id_map?.get(&k)?;
+                let fe_value = clone_id_map?.get(&(v?)).copied();
+                Some((*fe_key, fe_value))
+            })
+            .collect::<HashMap<Uuid, Option<Uuid>>>();
+    }
```

Comment:
Nitpick: Unwrap the Option of `clone_id_map` in the condition so that we don't need to do it every iteration.

```suggestion
    if sketch.clone.is_some()
        && let Some(clone_id_map) = clone_id_map
    {
        face_id_map = face_id_map
            .into_iter()
            .filter_map(|(k, v)| {
                let fe_key = clone_id_map.get(&k)?;
                let fe_value = clone_id_map.get(&(v?)).copied();
                Some((*fe_key, fe_value))
            })
            .collect::<HashMap<Uuid, Option<Uuid>>>();
    }
```